doc/spec/support-assessment.md

```diff
@@ -0,0 +1,237 @@
+A Bubbles extra manages and configures Prometheus and collects alerts
+for presentation.
+
+### Bubbles
+
+A Bubbles extra manages the Prometheus life cycle using the manager API
+to the orchestrator module. Handler for manager commands allow a
+service on the Prometheus container to poll configuration directly
+from the Bubbles extra.
+
+### Prometheus
+
+Containerized and run using the custom container feature of cephadm.
+This Prometheus is separate from the one shipped with cephadm.
+
+Our container runs once per cluster with its state stored on a RBD
+image with ext4. On start the containers receive a `ceph.conf` and
+credentials to interact with the cluster. Adapter service
+daemons running beside Prometheus facilitate runtime configuration
+from the Bubbles module reaching Prometheus.
+
+#### Target Discovery
+
+Discover targets by using [HTTP service discovery](https://prometheus.io/docs/prometheus/latest/http_sd/)
+
+Run a simple local service daemon that translates HTTP SD requests
+into RADOS manager commands handled by Bubbles.
+
+#### Rule Configuration
+
+Prometheus can't load rules from HTTP SD; it requires a local file and
+a signal or HTTP call to update.
+
+Similar to the target discovery use a local service daemon either
+polling Bubbles via manager commands or watch RADOS objects.
+
+#### Storage
+
+We use the default local file backed to persist metric data.
+Considering Ceph as the only storage option, we can either use CephFS
+or RBD. Exchanging one for the other later is trivial.
+
+Since we do not need concurrent file access and Prometheus actually discourages non-local
+filesystems (See warning in the [docs](https://prometheus.io/docs/prometheus/latest/storage/)), RBD is the
+best choice for now.
+
+#### The Case for Running a Second Prometheus
+
+Cephadm already runs Prometheus to aggregate metrics for the Dashboard
+and Grafana, since our use case is similar we could use that
+Prometheus as well. It is beneficial to run a second Prometheus with
+little drawbacks. Key arguments are:
+- We don't require long data retention, minimizing disk space
+  requirements
+- Running on shared store, we can migrate it between hosts giving us a
+  simple availability story. On fail over we would use freshness, but
+  maintain consistency. (The alternative would be multiple instances
+  with varying staleness)
+- Simplified configuration
+
+### Dashboard
+
+The dashboard part is out of scope for this proposal. Initially we have
+a simple JSON HTTP endpoint exporting support assessment data.
+
+### Custom Exporter
+
+We will add or extend existing Prometheus exporters where necessary.
+
+Where we create new exporters we may want to consider adding
+additional `/detail/$metric_name`-style endpoints. Bubbles could query
+them to receive additional information for human consumption not
+available from Prometheus directly.
+
+## Alternatives
+
+### Cephadm Configchecks
+
+Configuration checks build into the cephadm manager module. Tightly
```

Comment:
In ceph/ceph or aquarium?

Reply:
ceph/ceph. This thing here: https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/configchecks.py